client/src/components/ProfileEdit.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import { Row, Col, Input, Button } from 'react-materialize'
+import { request } from '../utils'
+import { Link } from 'react-router-dom'
+
+
+class ProfileEdit extends Component {
+
+    state = {
+        user_edit: {},
+        error_nick: "",
+        error_repeated: ""
+    }
+
+    componentDidMount() {
+        request('/profile/' + this.props.match.params.profile_id)
+        .then(json => {
+            this.setState({user_edit: json});
+            this.setState({
+                user_edit: {
+                    f_name: this.state.user_edit.first_name,
+                    l_name: this.state.user_edit.last_name,
+                    nick: this.state.user_edit.nickname,
+                    loc: this.state.user_edit.location,
+                    pass: this.state.user_edit.password
+                }
+            })
+        })
+    }
+
+    UpdateClick = (eve) => {
+        eve.preventDefault()
+
+        let msg_rep = ""
+
+        this.setState({ error_repeated: ""})
+
+        if (this.state.new_password === this.state.repeat_password) {
+            const data = {
+                "first_name": this.state.f_name,
+                "last_name": this.state.l_name,
+                "nickname": this.state.nickname,
+                "location": this.state.location,
+                "password": this.state.new_password,
+                "id": this.state.user_edit.id
+            }
+            request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
+            .then(data => {
+                if (data.msg !== "") {
+                    this.setState({ error_nick: data.msg })
+                }
+                else {
+                    this.setState({ error_nick: "" })
+                    this.props.history.push('/profile/' + this.props.match.params.profile_id);
+                    window.Materialize.toast("Profile Updated", 2500);
+                }
+            })
+        }
+        else {
+            msg_rep = "Incorrect repeated password"
+        }
+        this.setState({ error_repeated: msg_rep })
+    };
+
+    onChangeHandler = (e) => {
+        let { id } = e.currentTarget
+        this.setState({ [id]: e.currentTarget.value })
+    }
+
+    render() {
+      return (
+        <div>
+          <div>
+             <h1>Profile</h1>
+          </div>
+          <Row>
+            <Input
+              s={6}
+              error=""
+              id="f_name"
```

Comment:
I think it's better not to reduce the names of id.

Reply:
I think so too. You don't have any reason to type `f_name` instead of `first_name`.